rsocket-test/src/main/java/io/rsocket/test/TransportTest.java

```diff
@@ -215,14 +217,19 @@ default void requestChannel2_000_000() {
   @DisplayName("makes 1 requestChannel request with 3 payloads")
   @Test
   default void requestChannel3() {
-    Flux<Payload> payloads = Flux.range(0, 3).map(this::createTestPayload);
+    AtomicLong requested = new AtomicLong();
+    Flux<Payload> payloads =
+        Flux.range(0, 3).doOnRequest(requested::addAndGet).map(this::createTestPayload);
 
     getClient()
         .requestChannel(payloads)
-        .as(StepVerifier::create)
+        .as(publisher -> StepVerifier.create(publisher, 3))
         .expectNextCount(3)
         .expectComplete()
         .verify(getTimeout());
+
+    Assertions.assertThat(requested.get())
+        .isEqualTo(257L); // 257 because of eager behavior of limitRate
```

Comment:
@rstoyanchev for your attention. Here is the result of eager prefetch 1L (from the switchOnFirst + 256 extra)

Reply:
oh... found a bug because of skip does extra request 1 but it should not

Reply:
fixed. prefetch should be exactly 256